compiler/rustc_const_eval/src/interpret/discriminant.rs

```diff
@@ -26,7 +26,7 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
                 // No need to validate that the discriminant here because the
                 // `TyAndLayout::for_variant()` call earlier already checks the
                 // variant is valid.
-                let tag_dest = self.project_field(dest, tag_field)?;
+                let tag_dest = self.project_field(dest, tag_field.as_usize())?;
```

Comment:
`project_field` should probably take `FieldIdx`... but that can be a future PR.

Reply:
`project_field` is not used for arrays, we have `project_index` for that. (Those used to be one function but I refactored that a long time ago.)

Reply:
> If you say it'll work always here

I think it should, but I can't be sure without just trying it myself.^^

Reply:
>  I was thinking of https://doc.rust-lang.org/nightly/nightly-rustc/rustc_abi/enum.FieldsShape.html#method.offset and such.

Yeah that should be cleaned up to separate field offsets from array element offsets.